archive/qna_engine.py

```
@@ -255,3 +255,70 @@ def get_youtube_transcript(url: str) -> str:
         raise ValueError(f"Error fetching transcript: {str(e)}")
 
 
+def generate_mcq_math(self, topic, num=1, llm=None, response_model=None,
+                         prompt_template=None, custom_instructions=None, **kwargs):
+        if response_model is None:
+            parser = PydanticOutputParser(pydantic_object=MCQListMath)
```

Ruff

`MCQListMath` may be undefined, or defined from star imports

(F405)